core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.scheduler;
+
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import io.trino.metadata.Split;
+import io.trino.spi.exchange.ExchangeSourceHandle;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class TaskDescriptor
+{
+    private final int partitionId;
+    private final Multimap<PlanNodeId, Split> splits;
+    private final Multimap<PlanNodeId, ExchangeSourceHandle> exchangeSourceHandles;
+    private final NodeRequirements nodeRequirements;
+
+    public TaskDescriptor(
+            int partitionId,
+            Multimap<PlanNodeId, Split> splits,
+            Multimap<PlanNodeId, ExchangeSourceHandle> exchangeSourceHandles,
```

Comment:
Either splits or exchangeSourceHandles can be non-empty, right?

Can we make constructor private and create specific factory methods `createForSourceSplits` and `createForSourceExchange`?

Reply:
There could be tasks that have both (e.g.: join of a non bucketed table with a bucketed table)